core/src/main/java/me/mykindos/betterpvp/core/effects/listeners/effects/ProtectionListener.java

```diff
@@ -0,0 +1,52 @@
+package me.mykindos.betterpvp.core.effects.listeners.effects;
+
+import me.mykindos.betterpvp.core.effects.EffectManager;
+import me.mykindos.betterpvp.core.effects.EffectTypes;
+import me.mykindos.betterpvp.core.listener.BPvPListener;
+import me.mykindos.betterpvp.core.utilities.UtilMessage;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.entity.EntityDamageByEntityEvent;
+import org.bukkit.event.entity.EntityDamageEvent;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+@BPvPListener
+@Singleton
+public class ProtectionListener implements Listener {
+
+    private final EffectManager effectManager;
+
+    @Inject
+    public ProtectionListener(EffectManager effectManager) {
+        this.effectManager = effectManager;
+    }
+
+    @EventHandler
+    public void entityDamageEvent(EntityDamageEvent event) {
+        if (event.getEntity() instanceof Player player) {
+            if (effectManager.hasEffect(player, EffectTypes.PROTECTION)) {
```

Comment:
This stops all damage, including fall damage and mob damage. And presumably includes bosses (though probably will block them)

Reply:
ill probably tweak this when i properly implement the protection system (right now its never applied / command to remove it doesnt exist)